### PR TITLE
Implement rest parameter support for destructuring object patterns.

### PR DESCRIPTION
**Pull request by zherczeg**

### CODE REVIEW

File: jerry-core/vm/opcodes.c [hunk 1]
```diff
@@ -1687,6 +1678,137 @@ opfunc_assign_super_reference (ecma_value_t **vm_stack_top_p, /**< vm stack top
...
+
+    if (!(descriptor.flags & ECMA_PROP_IS_ENUMERABLE))
+    {
+      ecma_free_property_descriptor (&descriptor);
+      continue;
+    }
+
+    if ((descriptor.flags & ECMA_PROP_IS_VALUE_DEFINED) && !ECMA_OBJECT_IS_PROXY (source_object_p))
+    {
+      result = descriptor.value;
```
Comment (rerobika) [inline jerry-core/vm/opcodes.c:1783]: Shouldn't the property descriptor need to be released at this point?

Comment (zherczeg) [reply jerry-core/vm/opcodes.c:1783]: The reference is taken, so no need for it (an optimization).

Comment (rerobika) [reply jerry-core/vm/opcodes.c:1783]: I see now, OK.

**Review by rerobika — APPROVED** [review]
LGTM

**Review by dbatyai — APPROVED** [review]
LGTM

**Pull request merged**